scripts/webpack/getDefaultEnvironmentVars.js

```diff
@@ -0,0 +1,28 @@
+const getVariables = options => {
```

Comment:
What is this file for?

Reply:
I will add comments;

It's an "allow list" of variables to be fetched from the environment, along with default values if they aren't available.

When webpack comes along `process.env[name]`, the value will be replaced with the resolved values for these variables.

Values not in the allow list will not be replaced.

Reply:
> When webpack comes along process.env[name], the value will be replaced with the resolved values for these variables.

What if they're variables which really aren't set until runtime? Or am I misunderstanding how this works?

Reply:
it's a build-time-only find/replace for bundles. `process.env` is a node API for accessing environment variables; we never set those at runtime :) unless you're in a node environment, and even then, presumably you would produce a separate type of bundle for that.

Reply:
Still kind of confused, sorry. Did you actually hit errors related to any of these while running webpack? Though I guess really @layershifter or @miroslavstastny are better ones to ask about this since most of those variables are for v0 stuff.

Reply:
If I understand this correctly, node scripts read `process.env` directly, so the following will always work as `PORT` is read in node context:
```shell
PORT=8081 yarn gulp docs
```

As @dzearing wrote this only targets bundles - I don't think we use anything else then `NODE_ENV` there.

Reply:
Added comments